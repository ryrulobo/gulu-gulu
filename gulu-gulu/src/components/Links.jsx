import { Link, useLocation } from "react-router-dom";

export default function Links() {
  const links = [
    { url: "/search", text: "ðŸ”Ž All" },
    { url: "/images", text: "ðŸ“¸ Images" },
    { url: "/news", text: "ðŸ“° News" },
  ];

  const location = useLocation();

  return (
    <div className="flex sm:justify-around justify-between items-center mt-4">
      {links.map(({ url, text }) => (
        <Link to={url}>
          <p
            className={
              location.pathname === url
                ? "text-blue-700 border-b-2 border-blue-700 pb-1 pr-5"
                : "text-gray-700 pb-0 pr-5"
            }
          >
            {text}
          </p>
        </Link>
      ))}
    </div>
  );
}
