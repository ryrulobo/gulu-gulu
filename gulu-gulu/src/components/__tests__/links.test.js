import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import Links from "../Links";

afterEach(() => {
  cleanup();
});

const links = [
  { url: "/search", text: "ðŸ”Ž All" },
  { url: "/images", text: "ðŸ“¸ Images" },
  { url: "/news", text: "ðŸ“° News" },
];

test("Render links", () => {
  render(
    <BrowserRouter>
      <Links links={links} />
    </BrowserRouter>
  );
  const linksElement = screen.getByTestId("links");
  expect(linksElement).toBeInTheDocument();
  expect(screen.getByText("ðŸ”Ž All")).toBeInTheDocument();
  expect(screen.getByText("ðŸ“¸ Images")).toBeInTheDocument();
  expect(screen.getByText("ðŸ“° News")).toBeInTheDocument();
  expect(linksElement).toContainHTML("p");
  expect(linksElement.children).toHaveLength(3);
});

test("Matches snapshots", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Links links={links} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
