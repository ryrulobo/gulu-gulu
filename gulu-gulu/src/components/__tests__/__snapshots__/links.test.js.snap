// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  className="flex sm:justify-around justify-between items-center mt-4 xs:w-[10px] xsm:w-[10px]"
  data-testid="links"
>
  <a
    href="/search"
    onClick={[Function]}
  >
    <p
      className="text-gray-700 pb-0 pr-5"
    >
      ðŸ”Ž All
    </p>
  </a>
  <a
    href="/images"
    onClick={[Function]}
  >
    <p
      className="text-gray-700 pb-0 pr-5"
    >
      ðŸ“¸ Images
    </p>
  </a>
  <a
    href="/news"
    onClick={[Function]}
  >
    <p
      className="text-gray-700 pb-0 pr-5"
    >
      ðŸ“° News
    </p>
  </a>
</div>
`;
